github/resource_github_repository_autolink_reference.go

```diff
@@ -41,15 +39,22 @@ func resourceGithubRepositoryAutolinkReference() *schema.Resource {
 			"key_prefix": {
 				Type:        schema.TypeString,
 				Required:    true,
-				Description: "This prefix appended by a number will generate a link any time it is found in an issue, pull request, or commit",
 				ForceNew:    true,
+				Description: "This prefix appended by a number will generate a link any time it is found in an issue, pull request, or commit",
 			},
 			"target_url_template": {
 				Type:         schema.TypeString,
 				Required:     true,
+				ForceNew:     true,
 				Description:  "The template of the target URL used for the links; must be a valid URL and contain `<num>` for the reference number",
 				ValidateFunc: validation.StringMatch(regexp.MustCompile(`^http[s]?:\/\/[a-z0-9-.]*\/.*?<num>.*?$`), "must be a valid URL and contain <num> token"),
-				ForceNew:     true,
+			},
+			"is_alphanumeric": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    true,
+				Default:     true,
```

Comment:
[is_alphanumeric](https://docs.github.com/en/rest/repos/autolinks#create-an-autolink-reference-for-a-repository) is optional in the API and defaults to `true`